Gems/WhiteBox/Code/Source/SubComponentModes/EditorWhiteBoxTransformMode.cpp

```diff
@@ -602,9 +608,21 @@ namespace WhiteBox
             {
                 const AZ::Vector3 vertexLocalPosition =
                     (transformSelection->m_vertexPositions[vertexIndex++] - transformSelection->m_localPosition);
-                const AZ::Vector3 vertexPosition =
-                    (vertexLocalPosition * (AZ::Vector3::CreateOne() + (action.m_start.m_sign * action.LocalScaleOffset()))) +
-                    transformSelection->m_localPosition;
+                const AZ::Vector3 scale = [&action, &scaleType]
+                {
+                    switch (scaleType)
+                    {
+                    case ScaleType::Uniform:
+                        return AZ::Vector3(action.LocalScaleOffset().GetZ());
+                    case ScaleType::NonUniform:
+                        return action.LocalScaleOffset();
+                    default:
+                        break;
+                    }
+                    return AZ::Vector3();
```

Comment:
I know this line shouldn't get hit, but I think this will give an uninitialized Vector3, maybe use Vector3::CreateZero instead?